lib/js/src/manager/screen/choiceset/_ChoiceSetManagerBase.js

```diff
@@ -594,20 +623,85 @@ class _ChoiceSetManagerBase extends _SubManagerBase {
             .setKeypressMode(KeypressMode.RESEND_CURRENT_ENTRY);
     }
 
+    /**
+     * Clones a list of choice cells
+     * @param {ChoiceCell[]} originalList - A list of choice cells to be cloned
+     * @returns {ChoiceCell[]|null} - The cloned cell list
+     */
+    _cloneChoiceCellList (originalList) {
+        if (!Array.isArray(originalList)) {
+            return null;
+        }
+        return originalList.map((choiceCell) => choiceCell.clone());
+    }
+
     /**
      * Modifies the choices names depending on SDL version
      * @param {ChoiceCell[]} choices - The first list of choices
      * @returns {ChoiceCell[]} - A deep copy of the name modified choices
      */
     _getChoicesToBeUploadedWithArray (choices) {
+        const choicesClone = this._cloneChoiceCellList(choices);
         // If we're running on a connection < RPC 7.1, we need to de-duplicate cells because presenting them will fail if we have the same cell primary text.
         if (choices !== null && this._lifecycleManager.getSdlMsgVersion() !== null
                 && (this._lifecycleManager.getSdlMsgVersion().getMajorVersion() < 7
                 || (this._lifecycleManager.getSdlMsgVersion().getMajorVersion() === 7 && this._lifecycleManager.getSdlMsgVersion().getMinorVersion() === 0))) {
             // version if 7.0.0 or lower
-            this._addUniqueNamesToCells(choices);
+            this._addUniqueNamesToCells(choicesClone);
+        } else {
+            const strippedCellsClone = this._removeUnusedProperties(choicesClone);
+            this._addUniqueNamesBasedOnStrippedCells(strippedCellsClone, choicesClone);
         }
-        return choices.map(choice => choice.clone()); // deep copy
+        return this._preloadedChoices.reduce((filteredChoices, currentChoice) => {
+            return _ArrayTools.arrayRemove(filteredChoices, currentChoice);
```

Comment:
Just realized that a clone of choices is being passed into here, and arrayRemove does equality by reference comparison, which means that no matching choices would ever get removed. arrayRemove would have to be replaced by some other filter method that utilizes the equality check for choices instead.